src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -172,15 +172,26 @@ static NodeFunctionInvoker()
                 return;
             }
 
-            // if the function returned binding values via the function result,
-            // apply them to context.bindings
             var bindings = (Dictionary<string, object>)scriptExecutionContext["bindings"];
-            IDictionary<string, object> functionOutputs = functionResult as IDictionary<string, object>;
-            if (functionOutputs != null)
+            var returnValueBinding = outputBindings.SingleOrDefault(p => string.Compare(ScriptConstants.SystemReturnParameterBindingName, p.Metadata.Name) == 0);
```

Comment:
Would be nice to use the metadata to match on the `IsReturn` property so the logic is consistent across invokers.